token-swap/program/src/curve/calculator.rs

```diff
@@ -145,6 +148,16 @@ pub trait CurveCalculator: Debug + DynPack {
     fn allows_deposits(&self) -> bool {
         true
     }
+
+    /// Calculates the total value of the curve given the liquidity parameters.
+    ///
+    /// This is mostly useful for invariant testing the curves, to make sure
+    /// that value is not lost on any trade.
+    ///
+    /// The default implementation for this function gives the Uniswap invariant
+    fn total_value(&self, swap_token_a_amount: u128, swap_token_b_amount: u128) -> Option<U256> {
```

Comment:
Would this be more accurately called something like "normalized value" or "squared value" or "proportional_value"?

Reply:
I didn't want to drag in `PreciseNumber` and get the normalized value, but now that you mention this, there's no reason we should be able to take a square root since this is only used in tests.  Let's go with `normalized_value` in that case